src/UhdmAst.cpp

```diff
@@ -490,29 +430,107 @@ AstNode* get_value_as_node(vpiHandle obj_h, bool need_decompile = false) {
             valStr = "'d" + std::string(val.value.str);
         else if (val.format == vpiHexStrVal)
             valStr = "'h" + std::string(val.value.str);
-        if (int size = vpi_get(vpiSize, obj_h))
+        if (size != -1 && size != 0)
             valStr = std::to_string(size) + valStr;
-        auto* constp = new AstConst(make_fileline(obj_h), AstConst::StringToParse(), valStr.c_str());
-        valueNodep = constp;
-        break;
+        UINFO(7, "Obtained string is " << valStr << " for type " << val.format << std::endl);
+        return new AstConst(make_fileline(obj_h), AstConst::StringToParse(), valStr.c_str());
     }
     case vpiStringVal: {
         if (auto* s = val.value.str) valStr = std::to_string(*s);
         valStr.assign(val.value.str);
-        valueNodep = new AstConst(make_fileline(obj_h), AstConst::VerilogStringLiteral(),
+        return new AstConst(make_fileline(obj_h), AstConst::VerilogStringLiteral(),
                                   deQuote(make_fileline(obj_h), valStr));
-        break;
     }
     case 0: {
         UINFO(7, "No value; value format is 0" << std::endl);
-        return nullptr;
+        break;
     }
     default: {
         v3error("Encountered unknown value format " << val.format << std::endl);
         break;
     }
     }
-    return valueNodep;
+    return nullptr;
```

Comment:
In my opinion it would be better to throw error instead of returning nullptr. As I see, it can only happen when value format is 0, so it is a bug that needs to be reported.

Reply:
It's somewhat more complicated in this case. Value format can be 0 when node has no value. It is true that according to the standard each node should have its elaborated value, but we are using UHDM's folded model, so that won't be the case for many expression and parameter nodes.
That's why having no value -> no node -> nullptr returned here and then the caller decides whether this is an error or not (as opposed to unhandled, but present format, which indeed should be reported and fixed).

Reply:
It's somewhat more complicated in this case. Value format can be 0 when node has no value. It is true that according to the standard each node should have its elaborated value, but we are using UHDM's folded model, so that won't be the case for many expression and parameter nodes.
That's why having no value -> no node -> nullptr returned here and then the caller decides whether this is an error or not (as opposed to unhandled, but present format, which indeed should be reported and fixed).